### PR TITLE
Data base branch

### PR DESCRIPTION
**Pull request by DGMahendra**
I loaded the data fetched from the API into the api_db2 database. Below you can see the data loaded into each table.
country:
image
University:
image
person:
image

### CODE REVIEW

File: api_helper.py [hunk 1]
```diff
@@ -1,17 +1,99 @@
...
+def log_time(func):
+    def wrapper(*args, **kwargs):
+        start_time = time.time()
+        result = func(*args, **kwargs)
+        end_time = time.time()
+        logging.info(f"{func.__name__} took {end_time - start_time} seconds to complete.")
+        return result
+    return wrapper
+
 class APIFetch:
```
Comment (shreekanthbk22) [inline api_helper.py]: as this class contains functions like both API operations and database operations, change the name to appropriate

**Review by shreekanthbk22 — APPROVED** [review]

File: api_helper.py [hunk 2]
```diff
@@ -1,17 +1,99 @@
...
+            yield f"Error occurred: {e}"
+        except psycopg2.InterfaceError as e:
+            logging.error(f"InterfaceError occurred: {e}")
+
+    def _insert_generic(self, table_name, data):
+        try:
+            columns = data.keys()
+            values = tuple(data.values())
+            placeholders = ', '.join(['%s'] * len(values))
+            query = f"INSERT INTO {table_name} ({', '.join(columns)}) VALUES ({placeholders}) ON CONFLICT DO NOTHING RETURNING {table_name}_id"
```
Comment (shreekanthbk22) [inline api_helper.py:76]: instead of `ON CONFLICT DO NOTHING RETURNING` use `ON CONFLICT DO update`
because we want to maintain latest arrived data

**Review by shreekanthbk22 — APPROVED** [review]

File: api_helper.py (hunk 1 above)
Comment (DGMahendra) [reply api_helper.py]: I changed the class name to DataHandler

You can see below
image

File: api_helper.py (hunk 2 above)
Comment (DGMahendra) [reply api_helper.py:76]: I rewrite the code conflict code as below
image

**Pull request merged**